modin/pandas/test/utils.py

```diff
@@ -690,6 +690,20 @@ def modin_df_almost_equals_pandas(modin_df, pandas_df):
     )
 
 
+def try_modin_df_almost_equals_compare(df1, df2):
+    """Compare two dataframes as nearly equal if possible, otherwise compare as completely equal."""
+    # `modin_df_almost_equals_pandas` is numeric-only comparator
+    dtypes1, dtypes2 = map(
+        lambda df: dtype if is_list_like(dtype := df.dtypes) else [dtype], (df1, df2)
+    )
+    if all(is_numeric_dtype(dtype) for dtype in dtypes1) and all(
+        is_numeric_dtype(dtype) for dtype in dtypes2
+    ):
+        return modin_df_almost_equals_pandas(df1, df2)
+    else:
+        return df_equals(df1, df2)
```

Comment:
`df_equals` this function does not return anything.

```suggestion
        df_equals(df1, df2)
```